server/etcdserver/corrupt.go

```diff
@@ -585,6 +601,10 @@ func HashByRev(ctx context.Context, cc *http.Client, url string, rev int64) (*pb
 		if strings.Contains(string(b), mvcc.ErrFutureRev.Error()) {
 			return nil, rpctypes.ErrFutureRev
 		}
+	} else if resp.StatusCode == http.StatusPreconditionFailed {
+		if strings.Contains(string(b), "cluster ID mismatch") {
```

Comment:
It isn't good to compare a **hardcode** string. Please follow the same pattern as `mvcc.ErrCompacted`